Feature/config file

markhaur opened the pull request:
Added support for configuration file.
Added support for command line flag arguments.
Added goroutines for concurrency

markhaur committed:
added support for reading configuration file

markhaur committed:
added config file, command line params support, and go routines

markhaur committed:
refactored code with genius jarri-abidi. removed kong and used flag and other major improvements

jarri-abidi commented on modifier.go:
```diff
@@ -2,83 +2,100 @@ package main
+	var outputFile string
+	flag.StringVar(&configPath, "config", "./config.yml", "Path to configuration file.")
+	flag.DurationVar(&duration, "duration", 0, "Time modify duration. Must be negative for now.")
+	flag.StringVar(&inputFile, "i", "./input/Hz.Omer.S01E01.srt", "Path to input file.")
+	flag.StringVar(&outputFile, "o", "./output/Hz.Omer.S01E01.srt", "Path to output file.")
```
these should have empty defaults i think
```suggestion
	flag.StringVar(&inputFile, "i", "", "Path to input file.")
	flag.StringVar(&outputFile, "o", "", "Path to output file.")
```
otherwise a user can never load from a config file

markhaur replied:
great!

jarri-abidi commented on modifier.go:
```diff
@@ -2,83 +2,100 @@ package main
-		return
+func main() {
+	var configPath string
+	var duration time.Duration
```
could we call it offset instead of duration?

markhaur replied:
yeap. That would be much readable! Lemme do it

markhaur committed:
Update modifier.go

made empty defaults for input and output source files

Co-authored-by: Jarri Abidi <jarri1337@gmail.com>

jarri-abidi commented on modifier.go:
```diff
@@ -2,83 +2,100 @@ package main
+	var wg sync.WaitGroup
+	for _, cf := range configuration.Files {
+		wg.Add(1)
+		go func(iFile string, oFile string, mVal time.Duration) {
```
if we rename time to offset then this could be renamed to:
```suggestion
		go func(in, out string, offset time.Duration) {
```

jarri-abidi commented on modifier.go:
```diff
@@ -2,83 +2,100 @@ package main
+	wg.Wait()
+}
 
+func process(inputFile string, outputFile string, modifyValue time.Duration) error {
```
i think we should rename here too. inputFile would've been suitable for an os.File var. Since this is just a path, we should rename to:
```suggestion
func process(inputPath string, outputPath string, offset time.Duration) error {
```

markhaur committed:
Update modifier.go

renamed mVal to offset.

Co-authored-by: Jarri Abidi <jarri1337@gmail.com>

jarri-abidi commented on modifier.go:
```diff
@@ -2,83 +2,100 @@ package main
 	}
 
+	return nil
+
```
```suggestion
```

markhaur committed:
Update modifier.go

totally makes sense. Looks like I need to spend some quality time to learn naming variables.

Co-authored-by: Jarri Abidi <jarri1337@gmail.com>

markhaur committed:
Update modifier.go

i think we have removed only an empty line

Co-authored-by: Jarri Abidi <jarri1337@gmail.com>

jarri-abidi commented on modifier.go:
```diff
@@ -2,83 +2,100 @@ package main
 
 // currently supports subtraction of time
-func addValueToTime(time string, value int) string {
+func addValueToTime(time string, value time.Duration) string {
```
[optional] you plan to also support add in the future so using add/subtract in the name could be misleading:
```suggestion
func applyOffset(time string, offset time.Duration) string {
```

markhaur committed:
Update modifier.go

yeap. applyOffset makes much more sense

Co-authored-by: Jarri Abidi <jarri1337@gmail.com>

markhaur committed:
applied refactoring suggestion suggested by jarri-abidi

jarri-abidi commented on modifier.go:
```diff
@@ -2,83 +2,99 @@ package main
+}
 
+func process(inputPath string, outputPath string, offset time.Duration) error {
 	inpFile, err := os.Open(inputFile)
```
this one is optional/subjective.. 😄 
```suggestion
	inFile, err := os.Open(inputFile)
```
because opposite of out is in, not inp

jarri-abidi commented on config/Configuration.go:
```diff
@@ -0,0 +1,47 @@
+package config
```
it's almost always expected that in a pkg with only 1 file, that file should be named the exact same as the pkg
so in this case it should be config.go
also we avoid capital letters/camel case for file names as well as pkg names

markhaur replied:
I had to discuss this thing with you. It's good you pointed it out. lemme apply it.

markhaur committed:
changed filename from Configuration.go to config.go

jarri-abidi commented on config/config.go:
```diff
@@ -0,0 +1,47 @@
+	ModifyValue time.Duration
+}
+
+func GetConfiguration(path string) (*Configuration, error) {
```
```suggestion
func FromPath(path string) (*Configuration, error) {
```
then it would read as `config.FromPath("./somepath/config.yml")`

jarri-abidi replied:
Go teaches us to use pkg names as prefixes to describe our methods.. since it's pkg config, ofc one can guess we're going to do something about config
exactly what? the func name can describe that

markhaur replied:
Understood. Yes, it is much more descriptive. Even a non-tech can easily understand what is going on.

markhaur committed:
Update config/config.go

much descriptive

Co-authored-by: Jarri Abidi <jarri1337@gmail.com>

markhaur committed:
renamed variable names to much readable names

jarri-abidi commented on modifier.go:
```diff
@@ -2,83 +2,99 @@ package main
-
-	inpFile, err := os.Open(inputFile)
-
+	configuration, err := config.GetConfiguration(configPath)
```
this needs to be changed; wont compile now

markhaur replied:
Already did that. Other changes @jarri-abidi you made in method signatures also raised compile time errors. I've also resolved those.

markhaur committed:
Merge branch 'feature/config-file' of github.com:markhaur/srt-time-modifier into feature/config-file

markhaur committed:
refactored code

jarri-abidi reviewed (commented):
lgtm
